packages/web/src/pages/track-page/TrackPageProvider.tsx

```diff
@@ -100,8 +100,33 @@ type TrackPageProviderState = {
   source: string | undefined
 }
 
-class TrackPageProvider extends Component<
-  TrackPageProviderProps,
+// Create a functional wrapper component that uses the hook
+const TrackPageProviderWrapper = (props: TrackPageProviderProps) => {
+  const dispatch = useDispatch()
+  const params = parseTrackRoute(props.pathname)
+  const { data: track } = useTrackByParams(
+    params?.trackId
+      ? { id: params.trackId }
+      : {
+          handle: params?.handle ?? '',
+          slug: params?.slug ?? ''
+        }
```

Comment:
sad we can't just pass params

Reply:
yeah it's the id vs trackId. i could update the TrackScreen params to use trackId instead of id, what do you think?

Reply:
nbd I think

Reply:
working now 🔥